medical_prescription_state_verify/models/medical_prescription_order.py

```diff
@@ -71,13 +50,21 @@ def write(self, vals, ):
 
                 # Only allow state changes from self._ALLOWED_CHANGE_STATES
                 if vals.get('stage_id'):
-                    stage_id = self.env['%s.state' % self._name].browse(
+                    stage_id = self.env['base.kanban.stage'].browse(
                         vals['stage_id']
                     )
-                    if stage_id.type not in self._ALLOWED_CHANGE_STATES:
+                    if stage_id.name not in self._ALLOWED_CHANGE_STATES:
                         raise exceptions.ValidationError(_(
                             'You cannot move an Rx into this state after it'
                             ' has been verified. [%s]' % rec_id.name
                         ))
 
-            return super(MedicalPrescriptionOrder, self).write(vals)
+            super(MedicalPrescriptionOrder, self).write(vals)
+            if rec_id.stage_id.is_verified:
```

Comment:
This is outside of the record loop, but using the record. It's also causing two db writes & performing the same conditional twice. All the logic should be moved above & injected into vals before write.